src/components/screen/index.js

```diff
@@ -130,10 +130,26 @@ export default function Screen(props) {
 
         {tabs.length > 0 && (
           <Nav underline className="screen-navtabs">
-            {tabs.map((tab) => {
+            {tabs.map((tab, index) => {
               if (!tab) {
                 return null;
               }
+
+              // If tab has a component property, render that component
+              if (tab.component) {
+                const Component = tab.component;
+                return (
+                  <Component
+                    key={index}
+                    text={tab.text}
+                    to={tab.to}
+                    activeTab={activeTab}
+                    {...(tab.componentProps || {})}
+                  />
```

Comment:
@rsumilang is ther a reason not to continue using `tab.to` for the key? fyi react falls back to using the index when there is no key

also - does it actually need to know what the active tab is, or only if it is active?